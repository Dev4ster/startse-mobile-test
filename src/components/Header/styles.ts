import styled, { css } from 'styled-components/native';
import { SafeAreaView } from 'react-native-safe-area-context';

export const Wrapper = styled(SafeAreaView)`
  ${({ theme }) => css`
    height: 100px;
    justify-content: center;
    align-items: center;
    background-color: ${theme.colors.primary};
  `}
`;

export const Text = styled.Text`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.xlarge};
    font-weight: bold;
  `}
`;
