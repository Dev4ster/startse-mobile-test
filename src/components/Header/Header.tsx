import React, { useCallback, useEffect, useRef } from 'react';
import { Animated } from 'react-native';

import { Easing } from 'react-native-reanimated';
import * as S from './styles';
import Logo from '~/assets/image/logo.svg';

const Header = () => {
  const animationValue = useRef(new Animated.Value(0)).current;

  const onRenderAnimation = useCallback(
    () =>
      Animated.timing(animationValue, {
        toValue: 3000,
        duration: 2000,
        useNativeDriver: false,
        easing: Easing.elastic(3),
      }),
    [animationValue],
  );

  useEffect(() => {
    onRenderAnimation().start();
  }, [onRenderAnimation]);

  return (
    <S.Wrapper>
      <Animated.View
        style={{
          width: 200,
          opacity: animationValue.interpolate({
            inputRange: [0, 2000],
            outputRange: [0, 1],
          }),
          transform: [
            {
              translateX: animationValue.interpolate({
                inputRange: [0, 3000],
                outputRange: [-50, 0],
              }),
            },
          ],
        }}
      >
        <Logo />
      </Animated.View>
    </S.Wrapper>
  );
};

export default Header;
