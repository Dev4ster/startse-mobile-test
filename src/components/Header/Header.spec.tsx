import React from 'react';

import Header from './Header';
import { renderWithTheme } from '~/utils/test/helpers';

describe('<EmptyState />', () => {
  it('should be match snapshot', () => {
    const { toJSON } = renderWithTheme(<Header />);
    expect(toJSON()).toMatchSnapshot();
  });
});
