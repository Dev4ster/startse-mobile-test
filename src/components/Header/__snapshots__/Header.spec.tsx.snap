// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EmptyState /> should be match snapshot 1`] = `
<RNCSafeAreaView
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#0080ed",
        "height": 100,
        "justifyContent": "center",
      },
    ]
  }
>
  <View
    style={
      Object {
        "opacity": 0,
        "transform": Array [
          Object {
            "translateX": -50,
          },
        ],
        "width": 200,
      }
    }
  >
    <SvgMock />
  </View>
</RNCSafeAreaView>
`;
